lib/replace-in-file.js

```diff
@@ -60,33 +61,49 @@ module.exports = function replaceInFile(config, cb) {
   }
 
   //Initialize helper vars
-  var totalFiles = config.files.length;
+  var totalFiles = 0;
   var processedFiles = 0;
   var changedFiles = [];
 
   //No callback given? Perform sync operation
   if (!cb) {
     config.files.forEach(function(file) {
-      if (replaceSync(file, config.replace, config.with)) {
-        changedFiles.push(file);
-      }
+      //Expand globs
+      glob.sync(file, {nodir: true}).forEach(function(file) {
+        if (replaceSync(file, config.replace, config.with)) {
+          changedFiles.push(file);
+        }
+      });
     });
     return changedFiles;
   }
 
   //Otherwise, perform async operation
   config.files.forEach(function(file) {
-    replace(file, config.replace, config.with, function(error, hasChanged) {
-      if (error) {
-        return cb(error);
-      }
-      if (hasChanged) {
-        changedFiles.push(file);
+    //Expand globs
+    glob(file, {nodir: true}, function(err, files) {
+      if (err) {
+        return cb(err);
       }
-      processedFiles++;
-      if (processedFiles === totalFiles) {
-        cb(null, changedFiles);
+      //Glob doesn't error if no files match, so we do:
+      if (files.length === 0) {
+        return cb(new Error('No files match the pattern: ' + file));
       }
+      files.forEach(function(file) {
+        totalFiles++;
```

Comment:
Isn't `files` the total array of all files after the glob has been expanded though? So `files.length` should equal the total number of files, or am I missing something?

I'm hesitant to add more external dependencies, as I didn't have any up till now. But I guess since we are also including glob now I guess might as well, if it makes our life easier.

Feel free to try it and I'll have a look what you came up with.